test/Libraries/Microsoft.Extensions.DataIngestion.Tests/Chunkers/DocumentChunkerTests.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.Extensions.DataIngestion.Chunkers.Tests
+{
+    public abstract class DocumentChunkerTests
+    {
+        protected abstract IngestionChunker<string> CreateDocumentChunker(int maxTokensPerChunk = 2_000, int overlapTokens = 500);
+
+        [Fact]
+        public async Task ProcessAsync_ThrowsArgumentNullException_WhenDocumentIsNull()
+        {
+            var chunker = CreateDocumentChunker();
+            await Assert.ThrowsAsync<ArgumentNullException>(async () => await chunker.ProcessAsync(null!).ToListAsync());
```

Comment:
Let's verify the param name as well:

```suggestion
            await Assert.ThrowsAsync<ArgumentNullException>("chunks", async () => await chunker.ProcessAsync(null!).ToListAsync());
```